include/materials/material_utility.tcc

```diff
@@ -0,0 +1,271 @@
+//! Compute mean stress p (compression positive)
+inline const double mpm::materials::p(
+    const Eigen::Matrix<double, 6, 1>& stress) {
+  // Compute and return mean p
+  return (1. / 3. * (stress(0) + stress(1) + stress(2)));
+}
+
+//! Compute deviatoric stress
+inline const Eigen::Matrix<double, 6, 1> mpm::materials::deviatoric_stress(
+    const Eigen::Matrix<double, 6, 1>& stress) {
+
+  // Compute mean p
+  const double p = mpm::materials::p(stress);
```

Comment:
I found some paper use p=0.5*(sigma1+sigma3) and q=0.5*(sigma1-sigma3) for 2D. I think q is ok, and the current equation can represent the same result, but the p may be different. Sorry I'm not sure about this part, do you know which one is better?

Reply:
Hi @tianchiTJ those are MIT definition of mean and deviatoric stress. I suggest we stick to octohedral stress definition in 3D.

Reply:
OK, got it